src/js/components/tabs.js

```diff
@@ -0,0 +1,220 @@
+  _handleKeydown(event) {
+    const currentTab = event.target.closest(this.tabSelector);
+    // If not a tab, ignore
+    if (!currentTab) return;
+
+    // Create an array of all the focusable elements within the modal
+    const focusList = nodeListToArray(this.tabs);
+    const nextTab = focusList.indexOf(currentTab) + 1;
+    const prevTab = focusList.indexOf(currentTab) - 1;
+
+    switch (event.keyCode) {
+      case keyCodes.right || keyCodes.down: {
+
+        !focusList[nextTab] ? focusList[0].focus() : focusList[nextTab].focus();
+
+        break;
+      }
+      case keyCodes.down: {
+
+        !focusList[nextTab] ? focusList[0].focus() : focusList[nextTab].focus();
+
+        break;
+      }
+      case keyCodes.left: {
+
+        !focusList[prevTab] ? focusList[focusList.length - 1].focus() : focusList[prevTab].focus();
+
+        break;
+      }
+      case keyCodes.up: {
+
+        !focusList[prevTab] ? focusList[focusList.length - 1].focus() : focusList[prevTab].focus();
+
+        break;
+      }
+      case keyCodes.end: {
+
+        focusList[focusList.length - 1].focus();
+
+        break;
+      }
+      case keyCodes.home: {
+
+        focusList[0].focus();
+
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  }
+
+  /**
+   * This function is used to trigger the 'tabActivated' custom event,
+   * deactivate any other tab, and activate a tab/panel pair. It removes the
+   * panel's hidden attribute, adds the .rvt-tab-open class, sets the tab's
+   * aria-selected attribute to true. It also allows developers to setup a
+   * custom callback function.
+   * @param {Function} callback 
+   */
+  activateTab(tab, callback) {
+    // Trigger tabActivated custom event. This event is used to control the process of switching between tabs.
+
+    const activationEvent = dispatchCustomEvent(
+      'tabActivated',
+      tab,
+      {
+        id: tab.dataset.tabPanel,
+      }
+    );
+
+    if (!activationEvent) return;
+
+    const trigger = this.element.querySelector(`[${this.tabAttribute}="${tab.dataset.tabPanel}"]`);
+
+    this.panels.forEach(panel => {
+      if (panel.getAttribute(this.panelAttribute) !== tab.dataset.tabPanel && !panel.hasAttribute('hidden')) {
+        this.deactivateTab(panel);
+      }
+    });
+
+    // Activate the appropriate tab/panel pair
+    tab.removeAttribute('hidden');
+    tab.classList.add('rvt-tab-open');
+    trigger.setAttribute('aria-selected', 'true');
+    trigger.removeAttribute('tabindex');
+
+    if (callback && typeof callback === 'function') {
+      callback();
+    }
+  }
+
+  /**
+   * This function is used to trigger the 'tabDeactivated' custom event,
+   * and deactivates a tab/panel pair. It adds the hidden attribute, removes
+   * the .rvt-tab-open class, and sets the tab's aria-selected attribute to
+   * false. It also allows developers to setup a custom callback function.
+   * @param {Function} callback 
+   */
+  deactivateTab(tab, callback) {
```

Comment:
This method only ever gets called once inside the `activateTab()` method, which feels a little bit odd to me. I think we could get away with removing the important parts of this into the logic inside of `activateTab` and simplify things a bit.